packages/cra-template-microapp/template/src/search/subject.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import autobind from 'react-autobind';
+import { ObjectIcon } from '@infosight/elmer/dist/infrastructure';
+import productInterface from '../bootstrapper/productInterface';
+import { OBJECTS } from '../inventory/constants';
+import { mockInventory } from '../inventory/actionCreators';
+
+class BackendSearchSubject {
```

Comment:
Can you add a comment about why this is so complex? I think it was something to do with performance and worker threads and rate limiting.

 Or do you think this is enough of an improvement and we leave it at “do this here and it magically works”.

Reply:
I tried a few attempts to write a comment, but it sounded a lot like an apology rather than useful information.  I filled out a couple comments in other parts of the file and will let the overall file speak for itself.